### ISSUE TITLE
MenuStrip snap not working properly

### ISSUE THREAD

**Issue by stehlo** [acidb/mobiscroll#395]
MenuStrip Cards, when swiped, they don't slide for the full length of the strip on Amazon Fire (4th generation). They stop approximately in the middle of its length.

**Comment by stehlo** [comment]
The issue is present in v.3.0.0-beta5.

**Comment by stehlo** [comment]
I can't replicate this issue now. Either it is intermittent or it was a temporary problem.

**Comment by stehlo** [comment]
OK, I have to wake this issue up again, I am afraid.

On Xgody tablet (mentioned in [previous issue](https://github.com/acidb/mobiscroll/issues/394)), the MenuStrip Cards component is not reacting properly to swipe gestures and remains "stuck" in various positions – even out of the screen if manually moved out by finger (swipe).

**Comment by stehlo** [comment]
Addendum:
- Swipe right works fine.
- Swipe left is not working well at all.

**Comment by stehlo** [comment]
Just FYI:
The MenuStrip Navigation (type: options) is working fine in both directions on the same screen.

**Comment by stehlo** [comment]
This [left] swiping issue pertains also to Amazon Kindle Fire tablet.

**Comment by stehlo** [comment]
I have now more information about this issue:

Here are the steps to reproduce it:
- Place the Cards on in a div belonging on one tab (of MenuStrip Tabs).
- On app starting, the cards work just fine.
- Rotate the device and Cards are still fine.
- Change the tab (Cards are made hidden) and rotate the device.
- Return to the tab with Cards and see that the cards are already not fine.
- Rotate the device and Cards are still not fine.

Please note that on "deviceorientation", all the mobiscroll instances are correctly refreshed.

**Comment by dioslaska** [comment]
Thanks for the detailed steps to reproduce. Based on the description I think this will occur everywhere, regardless of enviroment.

The problem most probably is, that the Cards needs some dimensions for proper operations. On orientationchange these dimensions are automatically recalculated, but if the Cards are not visible (container div is invisible) , this won't work correctly.

The solution would be to call the [refresh](https://docs.mobiscroll.com/3-0-0_beta5/javascript/menustrip#!method-refresh) method of the Cards menustrip every time its tab becomes visible to force recalculation.

Let me know if this solves.

**Comment by stehlo** [comment]
This wouldn't explain why only left swipe is not OK.
I can confirm after multiple tests that refresh after both, screen orientation change and tab change simply doesn't help.
Something is not being recalculated properly.

**Comment by stehlo** [comment]
This issue persists even after the discovery of the problem with custom themes of your "custom download builder".

**Comment by stehlo** [comment]
OK, I have realised now, why your test app's cards are always fine...

It is necessary to use my settings in order to achieve the same problem at your end.

In stead of the following code:
```JavaScript
var cards = mobiscroll.menustrip('#cards', {
	display: 'inline',
	select: 'off',
	layout: 'fixed',
	itemWidth: 150
});
```

Use this one:
```JavaScript
var cards = mobiscroll.menustrip('#cards', {
	display: 'inline',
	select: 'off',
	layout: 1,
	snap: true
});
```

**Comment by dioslaska** [comment]
Using your code the issue is reproducible everywhere, even in the browser, in specific conditions:

  - menustrip is not visible at the time of the initialization
  - snap is set to true
  - itemWidth is not specified

The workaround in this case is to add `itemWidth: 1` to the init settings

**Comment by stehlo** [comment]
Thank you.

As far as I have seen, this workaround has other inevitable symptoms like displaying the the cards at a real 1px width. This needs to be corrected by rotating the device, which is utterly unacceptable for the users.

I developed a workaround by constructing the tab content on-the-fly, because I need to move on with the development.

However, I prefer to wait for a proper fix.

**Comment by dioslaska** [comment]
Fixed in 3.0.0-beta6
